File: Slavi_Stoyanchev_36_501219055_upr6/Calendar/src/com/company/PaymentDates.java
package com.company;

import java.time.DayOfWeek;
import java.time.LocalDate;

public class PaymentDates {

    public LocalDate CalculateFuturePaymentDate(LocalDate startingDate)
    {
        LocalDate tempDate = startingDate.plusDays(30);

        switch (tempDate.getDayOfWeek())
        {
            case SATURDAY: {
                tempDate = tempDate.plusDays(2);
                break;
            }
            case SUNDAY: {
                tempDate = tempDate.plusDays(1);
                break;
            }
        }
        return tempDate;
    }
}

File: PS_Slavi_Stoyanchev_36_501219055_upr5/vvps_exersice5/vvps_exersice5/PaymentDates.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace vvps_exersice5
{
    public class PaymentDates
    {
        public DateTime CalculateFuturePaymentDate(DateTime startingDate)
        {
            var tempDate = startingDate.AddDays(30);

            switch(tempDate.DayOfWeek)
            {
                case DayOfWeek.Saturday:
                    tempDate=tempDate.AddDays(2);
                    break;
                case DayOfWeek.Sunday:
                    tempDate = tempDate.AddDays(1);
                    break;
            }
            return tempDate;
        }
    }
}
